### PR TITLE
pinterest sources & 1 action

### PR DESCRIPTION
**Pull request by feyzullah**
### Triggers
- |&check;| New Board
- |&check;| New pin
- |&cross;| New follower: Could not find API endpoint for fetching followers in API `v5`
### Actions
- |&check;| Create pin 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3253"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/pinterest/pinterest.app.mjs [hunk 1]
```diff
@@ -1,11 +1,154 @@
...
   app: "pinterest",
-  propDefinitions: {},
+  propDefinitions: {
+    boardId: {
+      type: "string",
+      label: "Board ID",
+      description: "An ID identifying the board",
+      async options({ prevContext }) {
+        const pageSize = 25;
+        const { bookmark } = prevContext || {};
```
Comment (andrewjschuang) [inline components/pinterest/pinterest.app.mjs:13]: **-Minor-**

```suggestion
        const bookmark = prevContext?.bookmark;
```

File: components/pinterest/pinterest.app.mjs [hunk 2]
```diff
@@ -1,11 +1,154 @@
...
-      console.log(Object.keys(this.$auth));
+    async _getUrl(path) {
+      return `https://api.pinterest.com/v5${path}`;
+    },
+    _getHeaders(headers = {}) {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "user-agent": "@PipedreamHQ/pipedream v0.1",
```
Comment (andrewjschuang) [inline components/pinterest/pinterest.app.mjs:98]: **-Minor-**

```suggestion
        "User-Agent": "@PipedreamHQ/pipedream v0.1",
```

File: components/pinterest/pinterest.app.mjs [hunk 3]
```diff
@@ -1,11 +1,154 @@
...
+      };
+    },
+    async _makeRequest({
+      $,
+      path,
+      headers,
+      ...otherConfig
+    } = {}) {
+      const config = {
+        url: await this._getUrl(path),
```
Comment (andrewjschuang) [inline components/pinterest/pinterest.app.mjs:109]: Is there a reason for this method to be `async`?

File: components/pinterest/pinterest.app.mjs [hunk 4]
```diff
@@ -1,11 +1,154 @@
...
+        headers: this._getHeaders(headers),
+        ...otherConfig,
+      };
+      return axios($ ?? this, config);
+    },
+    async getBoards({ ...args } = {}) {
+      return await this._makeRequest({
```
Comment (andrewjschuang) [inline components/pinterest/pinterest.app.mjs:116]: `return await` [is not necessary](https://eslint.org/docs/latest/rules/no-return-await) unless it's inside a `try/catch` statement

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
@feyzullah thanks for the contribution!

Looking good, I've left some comments below.

**Review by andrewjschuang — APPROVED** [review]
Thanks for the changes, LGTM!

I've just left a comment about using `createdTime` instead of `Date.now()`

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Pinterest_3253.pdf](https://github.com/PipedreamHQ/pipedream/files/9029597/Pinterest_3253.pdf)

**Comment by feyzullah** [comment]
closing this one, created another PR because of merging issues [new PR link](https://github.com/PipedreamHQ/pipedream/pull/3376)

**Pull request closed without merging**